### PR TITLE
Review / Qual

### PR DESCRIPTION
**Pull request by aspangaro**
Review
Qual : Variables getNomUrl >> getObjectUrl
Qual : Renamed llx_c_tva into llx_c_vat
Qual : Renamed llx_tva into llx_vat
Qual : Update rate for example

### CODE REVIEW

File: htdocs/install/mysql/migration/3.6.0-3.7.0.sql [hunk 1]
```diff
@@ -1085,3 +1085,8 @@ CREATE TABLE llx_usergroup_extrafields (
 ALTER TABLE llx_usergroup_extrafields ADD INDEX idx_usergroup_extrafields (fk_object);
 
 ALTER TABLE llx_contrat ADD COLUMN model_pdf varchar(255) DEFAULT NULL AFTER note_public;
+
+-- Change on table c_tva
+ALTER TABLE llx_c_tva DROP INDEX uk_c_tva_id;
+ALTER TABLE llx_c_tva RENAME TO llx_c_vat;
+ALTER TABLE llx_c_vat ADD UNIQUE INDEX uk_c_vat_id(code);
```
Comment (hregis) [inline htdocs/install/mysql/migration/3.6.0-3.7.0.sql]: replace by:
ALTER TABLE llx_c_vat ADD UNIQUE INDEX uk_c_vat_id (fk_pays, taux, recuperableonly);

**Comment by aspangaro** [comment]
Thanks Regis, it's ok now !

**Comment by marcosgdf** [comment]
I'd say that getObjectUrl would be more apropiate than getNameUrl, because we are getting an URL that shows more info of that object

**Comment by aspangaro** [comment]
Hi @marcosgdf, You're the boss, other developers are OK ?

**Comment by marcosgdf** [comment]
@aspangaro wowowo, don't get me wrong!! I just wanted to say that maybe we could take the chance of renaming methods to consider if the name was the best one! But it is OK for me `getNameUrl`.

I really don't feel like in a higher position of any other developer and I don't want to. It is a shame that you noticed that :crying_cat_face:

**Comment by aspangaro** [comment]
You have a higher level than me like many other contributors in programming. You have my respect. Your advice is good. ;)

**Comment by marcosgdf** [comment]
Oh. I understood the opposite haha. Thanks ;D

I'll ask the dev list.

BTW: I don't know if you already did (the diff is too large to display), but if all objects extend from CommonObject you must create a public function getNomUrl that calls the new function (and mark it as deprecated) to increase compatibility.

**Comment by aspangaro** [comment]
In my last commit, I add this kind of function in each file class to increase backward compatibility :
-   Return clicable object (with eventually the picto) // Deprecated - For compatibility with external module
  *
-   @param  int     $withpicto      0=No picto, 1=Include picto into link, 2=Only Picto
-   @param  string  $option         Page link
-   @param  int     $maxlength      length max label
-   @return string                  String with URL
  */
  function getNomUrl($withpicto=0,$option='',$maxlength=0)
  {
    return $this->getObjectUrl($this->withpicto,$this->option,$this->maxlength);
  }

**Comment by marcosgdf** [comment]
To mark a function as deprecated you must do the following:

```
@deprecated Message pointing to new function
```

Also, if the objects extend from CommonObject you don't have to put the function in every single class but the parent class.

You could do some improvements to the documentation. I think that in English, picto is image and an appropiate description would be "Returns a HTML string containing a link (and an image) to the object"

**Comment by aspangaro** [comment]
Thanks for your help Marcos,

I can add this function in commonobject.class.php but the problem is each functions GetObjectUrl is different, variables as not the same in all class files. 

```
/**
 *  Returns a HTML string containing a link (and an image) to the object (kept for backward compatibility)
 *
 *  @param      int                $withpicto        0=No picto, 1=Include picto into link, 2=Only Picto        
 *  @param      string      $option     Page link
 *  @return       string                    String with URL
 *  @deprecated  Message pointing to new function
 */
function getObjectUrl($withpicto, $option)
{
    return $this->getNomUrl($withpicto,$option,'_public');
}
```

How can it be done?

**Comment by marcosgdf** [comment]
You have copied literally my message! :laughing: 

In the @deprecated message you have to tell the developer which function is now user. For example: @deprecated Use the new function getObjectUrl, but you have to put that in the old one, not in the new one.

To deal with that problem you should analyze all the signatures and behaviour and decide which is the more appropiate, if it is needed or not, etc and then create a common one.

Since you are creating a new function you can the order of the parameters and so.

**Comment by aspangaro** [comment]
I will be back ! ;)

**Pull request closed without merging**